staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -92,6 +92,11 @@ type ApplyOptions struct {
 	// not call the resource builder; only return the set objects.
 	objects       []*resource.Info
 	objectsCached bool
+
+	// Stores visited objects/namespaces for later use
+	// calculating the set of objects to prune.
+	visitedUids       sets.String
```

Comment:
Would these be useful for folks plugging in their own pruners?  If so, should they be public?

Reply:
Good point.